keys/keys.go

```diff
@@ -70,6 +70,42 @@ func MakeTableMetadataKey(namespaceID uint32, tableName string) proto.Key {
 	return k
 }
 
+// indexKeyWidth returns the width of all the columns forming the index key.
+func indexKeyWidth(columns ...[]byte) (width int) {
+	for _, column := range columns {
+		width += len(column)
+	}
+	return
+}
+
+// populateTableIndexKey populates the key passed in with the
+// order encoded values forming the index key.
+func populateTableIndexKey(key []byte, tableID, indexID uint32, columns ...[]byte) []byte {
+	key = append(key, TableDataPrefix...)
+	key = encoding.EncodeUvarint(key, uint64(tableID))
+	key = encoding.EncodeUvarint(key, uint64(indexID))
+	for _, column := range columns {
+		key = append(key, column...)
+	}
+	return key
+}
+
+// MakeTableIndexKey returns a primary or a secondary index key.
+func MakeTableIndexKey(tableID, indexID uint32, columns ...[]byte) proto.Key {
```

Comment:
Is `columns` intended to be column names are column values? If the latter, the name of the variable is confusing.